File: backend/pyspur/schemas/session_schemas.py
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel


class MessageBase(BaseModel):
    content: Dict[str, Any]


class MessageResponse(MessageBase):
    id: str
    session_id: str
    run_id: Optional[str]
    created_at: datetime
    updated_at: datetime


class SessionBase(BaseModel):
    pass


class SessionCreate(SessionBase):
    user_id: str


class SessionUpdate(SessionBase):
    pass


class SessionResponse(SessionBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime
    messages: List[MessageResponse]


class SessionListResponse(BaseModel):
    sessions: List[SessionResponse]
    total: int

File: frontend/src/types/api_types/sessionSchemas.ts
export interface MessageBase {
    content: Record<string, any>
}

export interface MessageResponse extends MessageBase {
    id: string
    session_id: string
    run_id?: string
    created_at: string
    updated_at: string
}

export interface SessionBase {
    workflow_id: string
}

export interface SessionCreate extends SessionBase {
    user_id: string
}

export interface SessionUpdate extends SessionBase {}

export interface SessionResponse extends SessionBase {
    id: string
    user_id: string
    created_at: string
    updated_at: string
    messages: MessageResponse[]
}

export interface SessionListResponse {
    sessions: SessionResponse[]
    total: number
}
